src/components/Dog.jsx

```diff
@@ -1,16 +1,33 @@
-import PropTypes from "prop-types";
-import "./Dog.css";
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import './Dog.css';
 
 const Dog = (props) => {
+    const [chipNum, setChipNum] = useState(props.chip);
+
+    //https://www.w3schools.com/js/js_random.asp
+    const getRndInteger = (min, max) => {
+      return Math.floor(Math.random() * (max - min)) + min;
+    };
+  
+    const createChip = () => {
+      setChipNum(getRndInteger(1000, 9999));
+      console.log();
+    };
+    
     return (
-      <li>
-        <h2>{props.name}</h2>
-        {props.chip ? (
-          <span>Chip #: {props.chip}</span>
-        ) : (
-          <button>add chip</button>
-        )}
-      </li>
+        <li>
+          <h2>
+            {props.name}
+            <button className="delete__dog">delete</button>
+          </h2>
+
+          {chipNum ? (
+            <span>Chip #: {chipNum}</span>
+          ) : (
+            <button onClick={createChip}>add chip</button>
```

Comment:
Personally, I'm a little uncomfortable with showing using what sounds more like a business logic function (createChip) as an event hanlder. We do show this elsewhere, but a comment reminding instructors to point out that really, React expects this to be an eventhandler (that accepts an event as its parameter), and that we can only pass this function because JS allows calling a function with more arguments than it expects (createChip doesn't expect any, but React is going to pass it event details anyway).

Alternatively, we could wrap this with small anonymous event handler function to really drive home the point that React is going to pass event details and we are choosing to ignore them.